bciers/apps/registration/app/components/operations/registration/OperationInformationPage.tsx

```diff
@@ -42,25 +42,31 @@ export const createOperationInformationSchema = (
   );
 
   const oneOfOptions = purposes.map((purpose) => {
+    const isRegulatedProducts =
+      purpose !== RegistrationPurposes.ELECTRICITY_IMPORT_OPERATION &&
+      purpose !== RegistrationPurposes.POTENTIAL_REPORTING_OPERATION;
     return {
-      // required: ["regulated_products"],
+      // This works:
+      // required: [isRegulatedProducts && "regulated_products"],
+      //
+      // This triggers validation error even when dynamic field isn't displayed:
+      required: ["regulated_products"],
       properties: {
```

Comment:
When testing this I uncommented the `minItems` and `required` field and initially thought everything was working as expected. However I noticed that the `required` validation would still be triggered if the field wasn't displayed.

To reproduce this:
- run the app
- go to `Register an Operation` form and on the first page `Operation Information` select `Potential Reporting Operation` as the `Registration Purpose` and select an Operation in the `Select your Operation` dropdown.
- Click `Save and continue`
- Nothing happens, you can open up your browser console and see the validation error for `regulated_products`

However if you try comment out this required and uncomment the conditional required above it works. I'm not used to this error and am not sure if it's due to this fix or something with our schema.

Reply:
It's an existing problem with the schema. The `map` adds `required: ["regulated_products"]` to all of the `oneOf` options, even the ones that don't actually have a `regulatedProducts` property, e.g.
![image](https://github.com/user-attachments/assets/1ff0bc77-2418-4040-a8a0-169703fe7097)

Reply:
Ah that makes sense, thanks so much for figuring this out. I wasn't completely confident I had fixed it.